test/utils/dummy_signer.ts

```diff
@@ -0,0 +1,160 @@
+        const confirmed = web3WrapperUtils.convertHexToNumber(
+            await this._getConfirmedTransactionCountAsync(this._signerAddress),
+        );
+        logger.info(`THE DIFF IS: ${pending} - ${confirmed} = `, pending - confirmed);
+        let nonceToUnstick = confirmed;
+        while (nonceToUnstick < pending) {
+            const txHash = await this.sendUnstickingTransactionAsync(
+                new BigNumber(HIGH_GAS_PRICE_FOR_UNSTICKING),
+                web3WrapperUtils.encodeAmountAsHexString(nonceToUnstick),
+            );
+            unstuckingTransactions.push(txHash);
+            nonceToUnstick++;
+        }
+
+        return unstuckingTransactions;
+    }
+
+    private async _sendTransactionAsync(
+        value: number,
+        nonce: string,
+        gasPrice: BigNumber,
+        expectedMinedInSec: number = EXPECTED_MINED_IN_S,
+    ): Promise<string> {
+        const ethereumTxnParams: PartialTxParams = {
+            from: this._signerAddress,
+            to: this._signerAddress,
+            value: web3WrapperUtils.encodeAmountAsHexString(value),
+            nonce,
+            chainId: TEST_RINKEBY_CHAIN_ID,
+            gasPrice: web3WrapperUtils.encodeAmountAsHexString(gasPrice),
+            gas: web3WrapperUtils.encodeAmountAsHexString(ETH_TRANSFER_GAS_LIMIT),
+        };
+        const ethTx = new EthereumTx(ethereumTxnParams);
+        ethTx.sign(this._privateKeyBuffer, true);
+        const txHash = ethTx.hash() as Buffer;
+        const txHashHex = `0x${txHash.toString('hex')}`;
+        logger.info('hashed: ', txHashHex);
+        logger.info('nonce: ', nonce);
+        const transactionEntity = new TransactionEntity({
+            hash: txHashHex,
+            status: TransactionStates.Unsubmitted,
+            nonce,
+            gasPrice: web3WrapperUtils.encodeAmountAsHexString(gasPrice),
+            metaTxnRelayerAddress: this._signerAddress,
+            expectedMinedInSec,
+        });
+        await this._transactionEntityRepository.save(transactionEntity);
+        const rawTx = `0x${ethTx.serialize().toString('hex')}`;
+        await this._web3Wrapper.sendRawPayloadAsync({
+            method: 'eth_sendRawTransaction',
+            params: [rawTx],
+        });
+        try {
+            await this._transactionEntityRepository.manager.transaction(async transactionEntityManager => {
+                transactionEntity.status = TransactionStates.Submitted;
+                await transactionEntityManager.save(transactionEntity);
+            });
+        } catch (err) {
+            // the TransacitonEntity was updated in the meantime. This will
+            // rollback the database transaction.
+            logger.warn('failed to store transaction with submitted status, rolling back', { err });
+        }
+        return txHashHex;
+    }
+
+    private async _getNonceAsync(senderAddress: string): Promise<string> {
```

Comment:
Move this to utils instead of duplicating.

Reply:
Could you expand on this?

I guess we can also just drop this and use `_getTransactionCountAsync` directly?